datafusion/expr/src/type_coercion/functions.rs

```diff
@@ -364,98 +366,73 @@ fn get_valid_types(
             return Ok(vec![vec![]]);
         }
 
-        let array_idx = arguments.iter().enumerate().find_map(|(idx, arg)| {
-            if *arg == ArrayFunctionArgument::Array {
-                Some(idx)
-            } else {
-                None
-            }
-        });
-        let Some(array_idx) = array_idx else {
-            return Err(internal_datafusion_err!("Function '{function_name}' expected at least one argument array argument"));
-        };
-        let Some(array_type) = array(&current_types[array_idx]) else {
-            return Ok(vec![vec![]]);
-        };
+        let mut fixed_size = None;
+        let mut large_list = false;
+        let mut element_types = Vec::with_capacity(arguments.len());
+        for (argument, current_type) in arguments.iter().zip(current_types.iter()) {
+            match argument {
+                ArrayFunctionArgument::Array => match current_type {
+                    DataType::FixedSizeList(field, size) => {
+                        match array_coercion {
+                            Some(ListCoercion::FixedSizedListToList) => (),
+                            None if fixed_size.is_none() => fixed_size = Some(*size),
+                            None if fixed_size == Some(*size) => (),
```

Comment:
if there are more than one fixed size list types like fixed(4), fixed(3).

The valid types should be `vec![fixed(4), fixed(3)]`

Reply:
If we found more than 1 fixed size list, return list might be the easier solution 🤔

Reply:
But return the `current_type` makes more sense.

Reply:
Oh, I see what you mean. I will fix that. `current_type` is not unified so we can't return that.